homeassistant/components/splunk/__init__.py

```diff
@@ -75,13 +58,11 @@ def setup(hass, config):
     name = conf.get(CONF_NAME)
     entity_filter = conf[CONF_FILTER]
 
-    if use_ssl:
-        uri_scheme = "https://"
-    else:
-        uri_scheme = "http://"
+    hec = http_event_collector(token, host, "json", name, port, use_ssl)
+    hec.SSL_verify = verify_ssl
 
-    event_collector = f"{uri_scheme}{host}:{port}/services/collector/event"
-    headers = {AUTHORIZATION: f"Splunk {token}"}
+    if not hec.check_connectivity():
+        _LOGGER.exception("Cannot connect to Splunk")
```

Comment:
Return False if it cannot connect to Splunk
```suggestion
        _LOGGER.exception("Error while trying to connect to Splunk")
        return False
```

Reply:
Will returning False disable the integration? I can see scenarios where this isn't desired, for example where Splunk and Home Assistant are on the same host, its restarted, and Splunk takes longer to startup than Home Assistant. The library used has an exponential backoff retry, so Splunk being offline for short periods is not a major issue. I did still think it was prudent to let the user know if there is a connectivity issue on start up though.

Reply:
> Will returning False disable the integration?

No, it will not disable the integration, however returning `False` indicates that initialization was unsuccessful since HA could not connect to Splunk. You could use that exponential back off retry if it is available.

Reply:
Returning false would skip setting up the listeners, essentially disabling the integration. It seems the library is also throwing its own errors when it has a connectivity issue. Seems redundant to log the same issue three times, so I am going to remove my warnings.

![](https://ba.id.au/errorlog.png)

I have raised a PR with the library owner to remove the warning on a successful connectivity check. https://github.com/georgestarcher/Splunk-Class-httpevent/pull/15